index.test-d.ts

```diff
@@ -51,6 +148,74 @@ expectType<string>(baseWithVoidAndNonVoidPlugins.bar);
 baseWithVoidAndNonVoidPlugins.unknown;
 
 const BaseWithOptionsPlugin = Base.plugin(withOptionsPlugin);
-const baseWithOptionsPlugin = new BaseWithOptionsPlugin();
+const baseWithOptionsPlugin = new BaseWithOptionsPlugin({
+  version: "1.2.3",
+});
 
 expectType<string>(baseWithOptionsPlugin.getFooOption());
+
+// Test depth limits of `.defaults()` chaining
+const BaseLevelFour = BaseLevelThree.defaults({ defaultFour: 4 });
+
+expectType<{
+  version: string;
+  defaultOne: string;
+  defaultTwo: number;
+  defaultThree: string[];
+  defaultFour: number;
+}>({ ...BaseLevelFour.defaultOptions });
+
+const baseLevelFour = new BaseLevelFour();
+
+// See the node on static defaults in index.d.ts for why defaultFour is missing
+// .options from .defaults() is only supported until a depth of 4
+expectType<{
+  version: string;
+  defaultOne: string;
+  defaultTwo: number;
+  defaultThree: string[];
+}>({ ...baseLevelFour.options });
+
+const BaseWithChainedDefaultsAndPlugins = Base
+  .defaults({
+    defaultOne: "value",
+  })
+  .plugin(fooPlugin)
+  .defaults({
+    defaultTwo: 0,
+  });
+
+const baseWithChainedDefaultsAndPlugins =
+  new BaseWithChainedDefaultsAndPlugins({
+    version: "1.2.3",
+  });
+
+expectType<string>(baseWithChainedDefaultsAndPlugins.foo);
```

Comment:
This is a tough one! It doesn't repro if either of the `.defaults({ ... })` are removed above... Tricky!

Reply:
Got it! It was the `this: Class` generic.

My leading hunch was that whatever information added the `foo: string` from the `.plugin(fooPlugin)` call was being abandoned by the `.defaults` calls going on top of each other. That meant the piece of the type that contained `foo` wasn't getting preserved.

That piece of type was the `this: Class` -> `& Class`. Using just one `<Class>` generic from the start meant new type information on new `this` scopes that had extra stuff added to them was being lost.